frontend/src/PublishMessageTile.tsx

```diff
@@ -0,0 +1,132 @@
+import * as React from "react";
+import './PublishMessageTile.css';
+import './UiTile.css';
+import './UiElements.css';
+import { EventBus, EventBusEventType } from "./EventBus";
+
+export interface SendTextForm {
+    message: string
+    platforms: Set<string>
+    eventBus: EventBus
+}
+
+/** Contains a form to publish new messages. */
+export class PublishMessageTile extends React.Component<SendTextForm, SendTextForm>{
+    constructor(props: SendTextForm, state: SendTextForm) {
+        super(props);
+        this.state = props;
+
+        this.state.eventBus.register(EventBusEventType.SELECTED_POST_CHANGED, (eventType, eventData?) => this.selectPost(eventData));
+    }
+
+    /** Event handler for when a post had been selected, e.g. via PublishedPostsTile.tsx */
+    private selectPost(eventData?: any) {
+        var platforms = new Set<string>();
+        platforms.add("" + eventData.platformId);
+        this.setState({
+            message: eventData.text,
+            platforms: platforms,
+            eventBus: this.state.eventBus
+        });
+    }
+
+    /** Send the request to the back end triggerin a post on the selected platforms. */
+    private send() {
+        fetch('http://localhost:8080/api/publish', {
+            method: 'POST',
+            headers: new Headers({
+                'Content-Type': 'application/json'
+            }),
+            credentials: 'include',
+            body: JSON.stringify({ message: this.state.message, platforms: Array.from(this.state.platforms.values()) })
+        })
+            .then(response => {
+                response.json();
+            })
+            .then(data => {
+                console.log("Result: " + data);
+                this.state.eventBus.fireEvent(EventBusEventType.REFRESH_POSTS);
+            });
+    }
+
+    private formTextAreaUpdated(event: React.ChangeEvent<HTMLTextAreaElement>) {
+        this.update(event.currentTarget.id, event.currentTarget.value);
+    }
+
+    private formInputFieldUpdated(event: React.ChangeEvent<HTMLInputElement>) {
+        this.update(event.currentTarget.id, event.currentTarget.value);
+    }
+
+    private update(id: string, value: string) {
+        if (id === "textMsg") {
+            this.setState({
+                message: value,
+                platforms: this.state.platforms,
+                eventBus: this.state.eventBus
+            });
+        }
+        if (id === "facebook" || id === "twitter") {
```

Comment:
When there will be more, you can write it like that for better readability:
```suggestion
        if (["facebook", "twitter"].includes(id)) {
```

You could also use
```suggestion
        if (id in ["facebook", "twitter"]) {
```

Reply:
done

Reply:
"facebook" in ["facebook", "twitter"] === false

Reply:
why do you need to do that explicitly? Wouldn't [`if(!(...))`](https://developer.mozilla.org/en-US/docs/Glossary/Type_coercion) suffice?